util/etcd/etcd.go

```diff
@@ -0,0 +1,368 @@
+	}
+	c := &Client{
+		timeout:       etcdTimeout,
+		backoffParams: etcdBackoff,
+		endpoint:      endpoint,
+		log:           logger,
+		etcdClient:    etcdClient,
+	}
+	return c, nil
+}
+
+// Members retrieves a list of etcd members
+func (c *Client) Members() (*[]Member, error) {
+	var lastError error
+	var resp *clientv3.MemberListResponse
+	err := wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		var err error
+		resp, err = c.etcdClient.MemberList(ctx)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get list of members for etcd cluster")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, errors.Wrap(lastError, "failed to get list of members for etcd cluster")
+	}
+
+	alarms, err := c.Alarms()
+	if err != nil {
+		return nil, err
+	}
+
+	members := make([]Member, len(resp.Members))
+	for i, m := range resp.Members {
+		newMember := pbMemberToMember(m)
+		for _, c := range *alarms {
+			if c.MemberID == newMember.ID {
+				newMember.Alarms = append(newMember.Alarms, c.Type)
+			}
+			members[i] = newMember
+		}
+	}
+
+	return &members, nil
+}
+
+// pbMemberToMember converts the protobuf representation of a cluster member
+// to a Member struct.
+func pbMemberToMember(m *etcdserverpb.Member) Member {
+	return Member{
+		ID:         m.GetID(),
+		Name:       m.GetName(),
+		PeerURLs:   m.GetPeerURLs(),
+		ClientURLs: m.GetClientURLs(),
+		IsLearner:  m.GetIsLearner(),
+		Alarms:     []AlarmType{},
+	}
+}
+
+// Member retrieves the Member information for a given peer
+func (c *Client) Member(peerURL string) (*Member, error) {
+	members, err := c.Members()
+	if err != nil {
+		return nil, err
+	}
+	for _, m := range *members {
+		if m.PeerURLs[0] == peerURL {
+			return &m, nil
+		}
+	}
+	return nil, errors.Errorf("etcd member %v not found", peerURL)
+}
+
+// Status returns the cluster status as known by the connected etcd server
+func (c *Client) Status() (*Status, error) {
+	var lastError error
+	var resp *clientv3.StatusResponse
+	err := wait.ExponentialBackoff(c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		var err error
+		resp, err = c.etcdClient.Status(ctx, c.endpoint)
+		cancel()
+		if err == nil {
+			return true, nil
+		}
+		c.log.Error(err, "failed to get etcd cluster status")
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, errors.Wrap(lastError, "failed to get etcd cluster status")
+	}
+
+	status := Status{
+		Leader:           resp.Leader,
+		Version:          resp.Version,
+		Errors:           resp.Errors,
+		IsLearner:        resp.IsLearner,
+		RaftIndex:        resp.RaftIndex,
+		RaftTerm:         resp.RaftTerm,
+		RaftAppliedIndex: resp.RaftAppliedIndex,
+		DbSizeInUse:      resp.DbSizeInUse,
+		DbSize:           resp.DbSize,
+	}
```

Comment:
Can we reuse the clientv3 StatusResponse as Status here? Seems lot of work duplicating structs